control/controlclient/direct.go

```diff
@@ -421,8 +421,19 @@ func (c *Direct) doLogin(ctx context.Context, opt loginOpt) (mustRegen bool, new
 	if res.StatusCode != 200 {
 		msg, _ := ioutil.ReadAll(res.Body)
 		res.Body.Close()
-		return regen, opt.URL, fmt.Errorf("register request: http %d: %.200s",
-			res.StatusCode, strings.TrimSpace(string(msg)))
+		var err error
+		if 400 <= res.StatusCode && res.StatusCode < 500 {
```

Comment:
How about we just don't send 400s from control's serveRegister then? And instead use RegisterResponse.Error and the line 452 path below?

(and we should fix that logging immediately before it to instead logf the error and not logf the "got response" until after the Error != "" path)

Reply:
I just don't like assuming that these 400s have any sort of defined body ... a plain text string? HTML? a JSON response?

I don't like just blindly shoving one of those 3 types into a MessageBox.

It'd be nice to know it's actually a text string, which RegisterResponse.Error already has.

Reply:
It's kinda gross that we return status 200, but also have an error to report.

Given that this code path already exists, I'll modify the control server to use that.